### PR TITLE
[ref] use one method to get boto client for aws bedrock

### LINKED ISSUE #11494: When bedrock's AKSK is left blank ( AWS environment configuration is automatically called), an error will be reported when using the rerank model
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.13.1

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

Deploy Dify on AWS EC2 and select Amazon Bedrock in Model Provider, Access Key and Secret Access Key Leave blank. At this time, the credentials owned by EC2 will be used by default to call bedrock. In the new version, when calling the newly added bedrock rerank model, an error will be reported without AKSK.
image

### ✔️ Expected Behavior

Successfully called bedrock rerank model

### ❌ Actual Behavior

[bedrock] Error: Partial credentials foundin explicit, missing: aws_secret_access_key

### PR DESCRIPTION
**Pull request by warren830**
# Summary

According to the issue #11494 , sometimes we might need to call bedrock without aksk, so I added a method to support this. And also modify other files within the directory.

Fixes #11494

> [!Tip]
> Close issue syntax: `Fixes #<issue number>` or `Resolves #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

# Screenshots
image

# Checklist

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

### CODE REVIEW

**Comment by crazywoola** [comment]
Please fix the lint

**Review by crazywoola — DISMISSED** [review]

File: api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import boto3
+from botocore.config import Config
+
+
+def get_bedrock_client(service_name, credentials=None):
+    client_config = Config(region_name=credentials["aws_region"])
+    aws_access_key_id = (credentials["aws_access_key_id"],)
+    aws_secret_access_key = credentials["aws_secret_access_key"]
+    if aws_access_key_id and aws_secret_access_key:
+        # 使用 AKSK 方式
```
Comment (yihong0618) [inline api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py]: better use English comments

File: api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        client = boto3.client(
+            service_name=service_name,
+            config=client_config,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+    else:
+        # 使用 IAM 角色方式
```
Comment (yihong0618) [inline api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py]: ditto

File: api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py (hunk 1 above)
Comment (warren830) [reply api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py]: fixed, thanks

File: api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py (hunk 2 above)
Comment (warren830) [reply api/core/model_runtime/model_providers/bedrock/get_bedrock_client.py]: fixed, thanks

**Review by crazywoola — APPROVED** [review]

**Pull request merged**